src/core/error_helpers.js

```diff
@@ -105,6 +110,110 @@ p5._friendlyFileLoadError = function (errorType, filePath) {
   report(message, errorInfo.method, FILE_LOAD);
 };
 
+/**
+* Validates Number type parameters
+* param  {String}               func    the name of the function
+* param  {String}               input   the input of the function
+* param  {Integer}              length  number of parameters to check
+*
+* return  {boolean}             message returns true if validated
+* return  {String}              err     the type of error
+* return  {Integer}             index   the location of error
+* return  {String}              message friendly err console log
+*
+* example:
+*  var a;
+*  ellipse(10,10,a,5);
+* console ouput:
+*  "It looks like ellipse received an empty variable in spot #2."
+*
+* example:
+*  ellipse(10,"foo",5,5);
+* console output:
+*  "ellipse was expecting a number for parameter #1,
+*           received "foo" instead."
+*/
+function validateNumParameters(func, args, length) {
+  var message;
+  if (arguments.length < 3) {
+    message = 'Missing input values for validating numeric parameters';
+    return [false, 'INIT_VALNUMPAR_FAIL', 0, message];
+  }
+  for (var p = 0; p < length; p++) {
+    var argType = typeof(args[p]);
+    message = '';
+    if ('undefined' === argType || null === argType) {
+      message = 'FES: It looks like ' + func +
+        ' received an empty variable in spot #' + p +
+        ' (zero-based index). If not intentional, this is often a problem' +
+        ' with scope: [link to scope].';
+      return [false, 'EMPTY_VAR', p, message];
+    } else if (argType !== 'number') {
+      message = 'FES: ' + func + ' was expecting a number' +
+        ' for parameter #' + p + ' (zero-based index), received ';
+      // Wrap strings in quotes
+      message += 'string' === argType ? '"' + args[p] + '"' : args[p];
+      message += ' instead.';
+      return [false, 'WRONG_TYPE', p, message];
+    }
+  }
+  return [true];
+}
+function validateParameters(func, args) {
```

Comment:
@lmccart 
this new function, `validateParameters()` is more generalized parameter validation function which looks up `doc/references/data.json` to check parameters.